### PR TITLE
fix(notification): fix some security vulnerabilities

### LINKED ISSUE #2000: [Bug]: the sign key of the notification channel is not encrypted
### ODC version

ODC 4.2.4

### OB version

none

### What happened?

The sign key of the notification channel is not encrypted

### What did you expect to happen?

it should be encrypted

### How can we reproduce it (as minimally and precisely as possible)?

create a channel with sign

### Anything else we need to know?

_No response_

### Cloud

_No response_

### PR DESCRIPTION
**Pull request by LuckyPickleZZ**
#### What this PR does / why we need it:

1. The sign key of the notification channel is not encrypted. This PR fixed it, when front-end send `null` value as sign, meaning it's already in metadb, we won't update this key.
2. Now only project's owner could test channel.
3. Support `ticketUrl` as event label key

#### Which issue(s) this PR fixes:

Fixes #2000

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/notification/helper/EventBuilder.java [hunk 1]
```diff
@@ -321,6 +331,34 @@ private <T> void resolveLabels(EventLabels labels, T task) {
...
+        String taskType = labels.get(TASK_TYPE);
+        String taskId = labels.get(TASK_ID);
+        Long organizationId;
+        if (task instanceof TaskEntity) {
+            organizationId = ((TaskEntity) task).getOrganizationId();
+        } else {
+            organizationId = ((ScheduleEntity) task).getOrganizationId();
+        }
+
+        String host = Optional.ofNullable(hostProperties.getOdcHost()).orElse(SystemUtils.getLocalIpAddress());
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/notification/helper/EventBuilder.java:351]: 1. lets' extract siteUrl cacluation logic to a separated component for reuse.

2. systemConfigService.queryByKeyPrefix, SystemConfigService may add a new method for queryByKey.

**Review by yizhouxw — APPROVED** [review]

File: server/odc-service/src/main/java/com/oceanbase/odc/service/notification/helper/EventBuilder.java (hunk 1 above)
Comment (LuckyPickleZZ) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/notification/helper/EventBuilder.java:351]: I'll do these in the next PR~

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**